src/collector.ts

```diff
@@ -17,23 +17,25 @@ import { autoScroll, fillForms } from './pptr-utils/interaction-utils';
 import { setupSessionRecordingInspector } from './session-recording';
 import { setUpThirdPartyTrackersInspector } from './third-party-trackers';
 import { clearDir } from './utils';
-export const collector = async ({
-    inUrl,
-    outDir = join(process.cwd(), 'bl-tmp'),
-    headless = true,
-    title = 'Blacklight Inspection',
-    emulateDevice = 'iPhone 13 Mini',
-    captureHar = true,
-    captureLinks = false,
-    enableAdBlock = false,
-    clearCache = true,
-    quiet = true,
-    defaultTimeout = 30000,
-    numPages = 3,
-    defaultWaitUntil = 'networkidle2',
-    saveBrowserProfile = false,
-    saveScreenshots = true,
-    blTests = [
+
+export type CollectorOptions = Partial<typeof DEFAULT_OPTIONS>;
+
+const DEFAULT_OPTIONS = {
+    outDir: join(process.cwd(), 'bl-tmp'),
+    title: 'Blacklight Inspection',
+    emulateDevice: 'iPhone 13 Mini' as keyof typeof KnownDevices,
```

Comment:
I actually had that previously, but went to `'iPhone 13 Mini' as keyof typeof KnownDevices` which you had in your example. But the whole reason I was doing this was to be able to pass custom devices in (puppeteer only has mobile device types). And your comment made me realize if I use Device I don't need a `viewport` and `userAgent` args, I can just have the caller create an object that matches that type to pass in. See example.js for what I mean by this.

Only downside here is we can't return emulateDevice in the output, since we no longer know the device name. But honestly returning that was silly. The caller already knows what type they requested. If we want it in the lambda API response we can add that back in in the lambda code.